src/verify/starknet.ts

```diff
@@ -67,16 +68,28 @@ export default async function verify(
   network: NetworkType = 'SN_MAIN',
   options: ProviderOptions = {}
 ): Promise<boolean> {
-  const contractAccount = new Contract(
-    ABI,
-    address,
-    getProvider(network, options)
-  );
+  try {
+    const contractAccount = new Contract(
+      ABI,
+      address,
+      getProvider(network, options)
+    );
+
+    if (sig.length < 2) {
+      throw new Error('Invalid signature');
+    }
 
-  await contractAccount.isValidSignature(getHash(data, address), [
-    sig[0],
-    sig[1]
-  ]);
+    const result = await contractAccount.is_valid_signature(
+      getHash(data, address),
+      sig.slice(-2)
+    );
 
-  return true;
+    return BigNumber.from(result).eq(BigNumber.from('370462705988'));
+  } catch (e: any) {
+    if (e.message.includes('Contract not found')) {
+      throw new Error('Contract not deployed');
+    }
+
+    throw e;
```

Comment:
Can we return `false` if error contains `INVALID_SIG`? 

this will help sequencer to return `invalid signature` if someone pass a invalid sig, right now we return `signature validation failed`